lib/jekyll/utils.rb

```diff
@@ -336,5 +326,34 @@ def duplicate_frozen_values(target)
         target[key] = val.dup if val.frozen? && duplicable?(val)
       end
     end
+
+    # Replace each character sequence with a hyphen.
+    #
+    # See Utils#slugify for a description of the character sequence specified
+    # by each mode.
+    private
+    def replace_character_sequence_with_hyphen(string, mode:)
```

Comment:
not a big fan of the long name `:replace_character_sequence_with_hyphen` but its a `private` method and the name expresses the method's intention.. so its okay I guess..

But, how about making the `:default` mode the *default* parameter?
```ruby
private
def replace_character_sequence_with_hyphen(string, mode: "default")
  replaceable_char =
    case mode
    when "raw"
      SLUGIFY_RAW_REGEXP
    when "pretty"
      # "._~!$&'()+,;=@" is human readable (not URI-escaped) in URL
      # and is allowed in both extN and NTFS.
      SLUGIFY_PRETTY_REGEXP
    when "ascii"
      # For web servers not being able to handle Unicode, the safe
      # method is to ditch anything else but latin letters and numeric
      # digits.
      SLUGIFY_ASCII_REGEXP
    else
      SLUGIFY_DEFAULT_REGEXP
    end

  string.gsub(replaceable_char, "-")
end
```
(and effectively reduce the total no.of `case` branches`)